scripts/tcs_startup.sh

```diff
@@ -50,34 +51,47 @@ start_avalon_components()
     python3 $LISTENER --bind_uri $LISTENER_PORT --lmdb_url $LMDB_URL &
     echo "Avalon Listener started"
 
+    sleep 5s
+    check_avalon_components
+
     if [ "$YES" != "1" ] ; then
         while true; do
-        echo "If you wish to exit the program, press y and enter"
-        read -t 5 yn
-        case $yn in
-            y ) stop_avalon_components;;
-            * ) echo " ";;
-        esac
+            echo "If you wish to exit the program, press y and enter"
+            read -t 5 yn
+            case $yn in
+                y ) stop_avalon_components;;
+                * ) echo " ";;
+            esac
         done
     fi
 }
 
+check_avalon_components()
+{
+    for i in $COMPONENTS ; do
+        pgrep -f "$i"
+        if [ $? != 0 ] ; then
+            echo "Terminating Avalon because component not running:"
+            echo "$LISTENER"
```

Comment:
Why are we only printing $LISTENER here? Is it for debugging purpose ?

Reply:
It is reporting an error. The listener terminated for some reason and it is reporting that fact

Reply:
My mistake. I fixed it by changing $LISTENER  to $i